src/main/java/org/opensearch/neuralsearch/processor/normalization/RRFNormalizationTechnique.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.normalization;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Locale;
+import java.util.Set;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.opensearch.neuralsearch.processor.CompoundTopDocs;
+
+import lombok.ToString;
+import org.opensearch.neuralsearch.processor.NormalizeScoresDTO;
+
+/**
+ * Abstracts calculation of rank scores for each document returned as part of
+ * reciprocal rank fusion. Rank scores are summed across subqueries in combination classes
+ */
+@ToString(onlyExplicitlyIncluded = true)
+public class RRFNormalizationTechnique implements ScoreNormalizationTechnique {
+    @ToString.Include
+    public static final String TECHNIQUE_NAME = "rrf";
+    public static final int DEFAULT_RANK_CONSTANT = 60;
+    public static final String PARAM_NAME_RANK_CONSTANT = "rank_constant";
+    private static final Set<String> SUPPORTED_PARAMS = Set.of(PARAM_NAME_RANK_CONSTANT);
+
+    private final int rankConstant;
+
+    public RRFNormalizationTechnique(final Map<String, Object> params, final ScoreNormalizationUtil scoreNormalizationUtil) {
+        scoreNormalizationUtil.validateParams(params, SUPPORTED_PARAMS);
+        rankConstant = getRankConstant(params);
```

Comment:
I am not sure whether this assignment would work. Considering rankingConstant is declared final at line 35.